build/tests/version-discovery.test.mjs

```diff
@@ -0,0 +1,150 @@
+import { describe, expect, it, beforeAll } from 'vitest';
+import { discoverVersions, getStarlightVersions } from '../version-discovery.mjs';
+
+describe('Version Discovery', () => {
+  let coreRepoPath;
+
+  beforeAll(() => {
+    coreRepoPath = process.env.CORE || process.env.PEPR_CORE_PATH;
+  });
+
+  describe('discoverVersions', () => {
+    it('should return versions and retired arrays', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const result = await discoverVersions(coreRepoPath, 2);
+      expect(result).toHaveProperty('versions');
+      expect(result).toHaveProperty('retired');
+      expect(Array.isArray(result.versions)).toBe(true);
+      expect(Array.isArray(result.retired)).toBe(true);
+    });
+
+    it('should include "latest" in versions', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const { versions } = await discoverVersions(coreRepoPath, 2);
+      expect(versions).toContain('latest');
+    });
+
+    it('should respect cutoff parameter', async () => {
+      if (!coreRepoPath) {
+        console.log('Skipping test: No CORE repository path provided');
+        return;
+      }
+
+      const cutoff = 2;
+      const { versions } = await discoverVersions(coreRepoPath, cutoff);
+      const versionCount = versions.filter(v => v !== 'latest').length;
+      expect(versionCount).toBeLessThanOrEqual(cutoff);
```

Comment:
A chance for a regression here could be if versionCount is N-1 when cutoff is N. Could this not catch a missing version? Then again, maybe this considers the case early in a project's lifecycle where there's only 1 version and we have a 2 version cutoff 🤷